stl/inc/atomic

```diff
@@ -418,21 +418,42 @@ void _Atomic_wait_direct(
 }
 #endif // _HAS_CXX20
 
-#if 1 // TRANSITION, ABI
+#if 1 // TRANSITION, ABI, GH-1151
 inline void _Atomic_lock_acquire(long& _Spinlock) noexcept {
-    while (_InterlockedExchange(&_Spinlock, 1)) {
-        _YIELD_PROCESSOR();
+#if defined(_M_IX86) || defined(_M_X64)
+    // Algorithm from Intel(R) 64 and IA-32 Architectures Optimization Reference Manual, September 2019
+    // Example 2-4. Contended Locks with Increasing Back-off Example - Improved Version, page 2-21
+    int _Current_backoff   = 1;
+    const int _Max_backoff = 64;
+    while (_InterlockedExchange(&_Spinlock, 1) != 0) {
+        while (__iso_volatile_load32(&reinterpret_cast<int&>(_Spinlock)) != 0) {
+            for (int i = _Current_backoff; i != 0; --i) {
+                _mm_pause();
+            }
+            _Current_backoff = _Current_backoff < _Max_backoff ? _Current_backoff << 1 : _Max_backoff;
+        }
     }
+#elif defined(_M_ARM) || defined(_M_ARM64)
+    while (/* TRANSITION, GH-1133, _InterlockedExchange_acq*/ _InterlockedExchange(&_Spinlock, 1) != 0) {

```

Comment:
Could we please move the comment out of the code, a la:
```suggestion
    while (_InterlockedExchange(&_Spinlock, 1) != 0) { /* TRANSITION, GH-1133: _InterlockedExchange_acq
```

Reply:
Will do that.
Wanted traditional `#if 0`, but clang format is very confused about mismatched braces in preprocessor blocks.